pkg/controllers/dnspolicy/dns_helper_test.go

```diff
@@ -378,175 +347,41 @@ func TestDNS_findMatchingManagedZone(t *testing.T) {
 			Name:  "no managed zones for host give error",
 			Host:  "sub.domain.test.example.co.uk",
 			Zones: []v1alpha1.ManagedZone{},
-			Assert: func(t *testing.T, zone *v1alpha1.ManagedZone, subdomain string, err error) {
-				if err == nil {
-					t.Fatalf("expected error, got %v", err)
-				}
-			},
-		},
-		{
-			Name: "should not match when host and zone domain name are identical",
-			Host: "test.example.com",
-			Zones: []v1alpha1.ManagedZone{
-				{
-					ObjectMeta: v1.ObjectMeta{
-						Name:      "test.example.com",
-						Namespace: "test",
-					},
-					Spec: v1alpha1.ManagedZoneSpec{
-						DomainName: "test.example.com",
-					},
-				},
-			},
-			Assert: func(t *testing.T, zone *v1alpha1.ManagedZone, subdomain string, err error) {
```

Comment:
I think you are moving a test that was on main.

Very difficult to tell from the diff since so much has changed, but there should be a test "should not match when host and zone domain name are identical" in TestDNS_findMatchingManagedZone

Reply:
fixed